raster-foundry/raster-foundry
# Add Scopes to API Routes

## Overview

This PR adds scopes to our API routes. Additionally, I edited some of the canned scopes. After adding all of (what I think) are the required scopes I was a little confused by some of the breakdown in the scopes and did roughly the following:
 -  I got rid of the "Multiplayer" scopes and added "ReadPermissions" and "Share" to the CRUD scope creation helper. I can't think of any cases where an RF user would be able to create/read/update/delete something and then later not be able to read permissions on it. This will not be true for GroundWork, but we'll need I think a specific canned policy for that
 - I reordered things more generally in Scope.scala to be alphabetical, except with the _big_ RF canned scopes to be at the bottom

### Checklist


### Notes

I updated the [spreadsheet](https://docs.google.com/spreadsheets/d/1CD1dK47UpVcqgBsl3d9Ol0Fk5po9LKmKnTNXZbPHcDg/edit#gid=115836025) for scopes and breakdowns. To the best of my ability this is what I followed when applying scopes to routes and creating canned policies.

I'm not sure what we should do with the `RasterFoundryTeamsAdmin` scope included in the migration for scopes. I couldn't identify (see second sheet from [here](https://docs.google.com/spreadsheets/d/1CD1dK47UpVcqgBsl3d9Ol0Fk5po9LKmKnTNXZbPHcDg/edit#gid=115836025)) a case where a regular user and team admin would differ since any user can create a team within their organization.

I opened two issues for things not included in this PR
 - [one](https://github.com/raster-foundry/annotate/issues/531) for encoding numerical limits in scope checking
 - [one](https://github.com/raster-foundry/annotate/issues/532) for adding integration tests that we can run in CI

## Testing Instructions

- Make sure migrations are up-to-date
- Assemble server
- Browse as dev user
- Run `update users set scopes = '"platformUser"'::jsonb where email = 'dev@rasterfoundry.com';`, restart/stop _everything_ `docker-compose stop`
- Try to go to the platforms page: http://localhost:9091/admin/platform/31277626-968b-4e40-840b-559d9c67863c and you shouldn't be able to because there is a `403` from a platform admin route
- Review some of the changes to canned policies, make sure they make sense

Closes #5261
